keras_cv/callbacks/pycoco_callback.py

```diff
@@ -95,20 +92,24 @@ def boxes_only(images, boxes):
         ground_truth["height"] = [tf.tile(tf.constant([height]), [total_images])]
         ground_truth["width"] = [tf.tile(tf.constant([width]), [total_images])]
 
-        ground_truth["num_detections"] = [gt_boxes.row_lengths(axis=1)]
-        ground_truth["boxes"] = [gt_boxes.to_tensor(-1)]
-        ground_truth["classes"] = [gt_classes.to_tensor(-1)]
-        box_pred = bounding_box.convert_format(
-            box_pred, source=self.bounding_box_format, target="yxyx"
+        ground_truth["num_detections"] = [boxes.row_lengths(axis=1)]
+        ground_truth["boxes"] = [boxes.to_tensor(-1)]
+        ground_truth["classes"] = [classes.to_tensor(-1)]
+
+        y_pred = bounding_box.convert_format(
+            y_pred, source=self.bounding_box_format, target="yxyx"
         )
 
-        predictions = {}
+        y_pred = bounding_box.to_ragged(y_pred)
```

Comment:
Same as above -- we always return dense tensors with `num_detections`, so we don't need to run any ragged conversion here

Reply:
Doing this makes us support both Ragged/Dense predictions, and it makes them consistent as well.  Callbacks also run on CPU so no harm done to TPU compat.